lib/JsonLdDocumentLoader.js

```diff
@@ -15,7 +16,7 @@ export class JsonLdDocumentLoader {
     if(!_isObject(document)) {
       throw new TypeError('The second parameter (document) must be an object.');
     }
-    this.documents.set(url, document);
+    this.documents.set(url, klona(document));
```

Comment:
Would `structuredClone` work instead?

https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

```suggestion
    this.documents.set(url, structuredClone(document));
```